src/scene/vcFolder.cpp

```diff
@@ -315,7 +315,7 @@ void vcFolder::HandleImGui(vcState *pProgramState, size_t *pItemID)
         ImGui::EndPopup();
       }
 
-      if (pNode->itemtype != vdkPNT_Folder && pSceneItem->GetWorldSpacePivot() != udDouble3::zero() && ImGui::IsMouseDoubleClicked(0) && ImGui::IsItemHovered())
+      if (pNode->itemtype != vdkPNT_Folder /*&& pSceneItem->GetWorldSpacePivot() != udDouble3::zero()*/ && ImGui::IsMouseDoubleClicked(0) && ImGui::IsItemHovered())
```

Comment:
I think this should just check if the vcSLS is loaded?

Reply:
Perhaps we need to check this anyway, although I think the issue comes from the fact that every item derived from vcSceneItem has a World matrix and a pivot; even items where this doesn't really make sense, folders for instance. That would suggest it's in the '3D' scene and can potentially interact with other objects in the scene.

Reply:
Possible fix is a deferred action... seems quite often my solutions come in the form of deferred actions, perhaps someone should come up with a framework for deferring actions by either frame count, time delta or a function pointer that you'd pass in. the vcSLS loaded check addresses the immediate issue but I expect QA will then log that their expectations still aren't met

Reply:
not sure, was this comment correct? 
```
Damian Madden
commented Nov 5, 2019
 
@Paul Fox To clarify, this is the case regardless of a UDS being in the scene or not, and because VSM files aren't located the position is equal to udDouble3::zero, which is a sentinel for the scene item being not loaded or otherwise in a state from which no world space pivot can be determined... I could fix by changing the default position to non-zero, possibly loading a VSM at the camera position? or immediately in front of the camera
```
A quick look turned up nothing that sets it for a new model

Reply:
Right so I think the solution will be choosing some non-zero origin for vcPolygonModel_CreateFromVSMFInMemory & vcPolygonModel_CreateFromRawVertexData

Reply:
or some other sentinel for unset polygonmodels

Reply:
Ok added a more elegant solution.